File: building-a-wallet/python/step3_tx_history.py
import sys
from solana.rpc.api import Client
from solders.pubkey import Pubkey

ENDPOINT = "https://your-endpoint.rpcpool.com"
TOKEN = "your-token"

def get_transaction_history(wallet_address: str, limit: int = 10):
    client = Client(f"{ENDPOINT}/{TOKEN}")
    pubkey = Pubkey.from_string(wallet_address)

    sigs = client.get_signatures_for_address(pubkey, limit=limit)
    print(f"Found {len(sigs.value)} transactions")

    for sig_info in sigs.value:
        status = "failed" if sig_info.err else "ok"
        print(f"{str(sig_info.signature)[:20]}... slot={sig_info.slot} status={status}")

wallet = sys.argv[1] if len(sys.argv) > 1 else "YourWalletAddressHere"
get_transaction_history(wallet)

File: building-a-wallet/typescript/step3-tx-history.ts
import {
  Connection,
  PublicKey,
  ParsedTransactionWithMeta,
} from "@solana/web3.js";

const ENDPOINT = "https://your-endpoint.rpcpool.com";
const TOKEN = "your-token";

async function getTransactionHistory(
  walletAddress: string,
  limit: number = 10
): Promise<ParsedTransactionWithMeta[]> {
  const connection = new Connection(`${ENDPOINT}/${TOKEN}`);
  const pubkey = new PublicKey(walletAddress);

  const signatures = await connection.getSignaturesForAddress(pubkey, { limit });
  console.log(`Found ${signatures.length} transactions`);

  const transactions = await connection.getParsedTransactions(
    signatures.map((s) => s.signature),
    { maxSupportedTransactionVersion: 0 }
  );

  for (const tx of transactions) {
    if (!tx) continue;
    const sig = tx.transaction.signatures[0];
    const slot = tx.slot;
    const status = tx.meta?.err ? "failed" : "ok";
    const fee = tx.meta?.fee ?? 0;
    console.log(`${sig.slice(0, 20)}... slot=${slot} status=${status} fee=${fee}`);
  }

  return transactions.filter(Boolean) as ParsedTransactionWithMeta[];
}

const wallet = process.argv[2] ?? "YourWalletAddressHere";
getTransactionHistory(wallet).catch(console.error);
